source/type/type.cpp

```diff
@@ -506,6 +506,21 @@ std::vector<idx_t> MetaVehicleJourney::get(Type_e type, const PT_Data& data) con
     return result;
 }
 
+bool MetaVehicleJourney::is_already_impacted_by(const boost::shared_ptr<disruption::Impact>& impact) {
+    auto find_impact = [&](const boost::weak_ptr<nt::disruption::Impact>& i) {
+            auto spt = i.lock();
+            return (spt) ? spt == impact : true;
```

Comment:
if spt is nullptr, you  want to return false, no? or you want to do your test on impact?